tensorflow/core/kernels/data/experimental/sampling_dataset_op_test.cc

```diff
@@ -0,0 +1,847 @@
+      gtl::InlinedVector<TensorValue, 4>* const inputs,
+      std::unique_ptr<OpKernelContext>* context) {
+    TF_RETURN_IF_ERROR(CheckOpKernelInput(*op_kernel, *inputs));
+    TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
+    return Status::OK();
+  }
+};
+
+// TODO(frreiss): Remove this once #31344 goes in and RangeDatasetParams is
+// defined in dataset_test_base.h
+class LocalRangeDatasetParams : public DatasetParams {
+ public:
+  LocalRangeDatasetParams(int64 start, int64 stop, int64 step,
+                          DataTypeVector output_dtypes,
+                          std::vector<PartialTensorShape> output_shapes,
+                          string node_name)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes),
+                      std::move(node_name)),
+        start(CreateTensor<int64>(TensorShape({}), {start})),
+        stop(CreateTensor<int64>(TensorShape({}), {stop})),
+        step(CreateTensor<int64>(TensorShape({}), {step})) {}
+
+  Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) override {
+    *inputs = {TensorValue(&start), TensorValue(&stop), TensorValue(&step)};
+    return Status::OK();
+  }
+
+  Tensor start;
+  Tensor stop;
+  Tensor step;
+};
+
+class SamplingDatasetParams : public DatasetParams {
+ public:
+  SamplingDatasetParams(float rate, int64 seed, int64 seed2, int64 start,
+                        int64 stop, int64 step, DataTypeVector output_dtypes,
+                        std::vector<PartialTensorShape> output_shapes,
+                        string node_name)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes),
+                      std::move(node_name)),
+        rate(CreateTensor<float>(TensorShape({}), {rate})),
+        seed(CreateTensor<int64>(TensorShape({}), {seed})),
+        seed2(CreateTensor<int64>(TensorShape({}), {seed2})),
+        range_dataset_params(start, stop, step, {DT_INT64},
+                             {PartialTensorShape({})}, "") {}
+
+  Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) override {
+    if (input_dataset.NumElements() == 0 ||
+        input_dataset.dtype() != DT_VARIANT) {
+      return tensorflow::errors::Internal(
+          "The input dataset is not populated as the dataset tensor yet.");
+    }
+    *inputs = {TensorValue(&input_dataset), TensorValue(&rate),
+               TensorValue(&seed), TensorValue(&seed2)};
+    return Status::OK();
+  }
+
+  // Static parameters of the kernel
+  Tensor rate;
+  Tensor seed;
+  Tensor seed2;
+
+  // Parameters of the sequence of numbers that will serve as the dynamic input
+  // of the kernel.
+  LocalRangeDatasetParams range_dataset_params;
+
+  // RangeDataset kernel wrapped in a variant tensor. Initialized by the test
+  // case itself because the MakeRangeDataset() method requires an instance of
+  // DatasetOpsTestBase.
+  Tensor input_dataset;
+};
+
+SamplingDatasetParams OneHundredPercentSampleDataset() {
```

Comment:
Values that do not change across the different test cases (seed, seed2, start, step) should not be an argument. `stop` can be replaced with `num_elements`.

Reply:
Thanks for that feedback. I've pushed commit 02d9d92, which replaces the `seed`, `seed2`, `start`, and `step` parameters with constants and renames `stop` to `num_elements`.